### DISCUSSION TITLE
New add-on: Styles Color Shifter

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#1516] [Show and tell]
Hi all,
I created an add-on that "shifts" the colors of the template. It requires **Freeplane v1.11.8-pre6 or newer** to work.

Repository: [StylesColorShifter](https://github.com/EdoFro/Freeplane_StylesColorShifter)

the installable is in its [releases](https://github.com/EdoFro/Freeplane_StylesColorShifter/releases) page

-----

![image](https://github.com/freeplane/freeplane/assets/35700575/f9f923e6-f2c9-4b13-8857-38bc7e4f3217)

-----

### Shifting colors using the dialog

https://github.com/EdoFro/Freeplane_StylesColorShifter/assets/35700575/07636966-36b3-4df4-8a9a-1ef55a4e765c

### Random shift of colors

https://github.com/EdoFro/Freeplane_StylesColorShifter/assets/35700575/092d7fb8-2184-4c8b-b8f3-f250d54ba3d9

Hope it helps,

edo

**Comment by i-plasm** [comment]
@EdoFro It's great that API now allows programmatic styling like that. It would be really useful to also allow setting the CSS of any node style. It would open many cool and useful possibilities for scripting and more user-friendly ways of styling. Thank you!

**Reply by Foxxsayy** [reply]
It looks like this shades child nodes a shade darker or lighter than the parent. If I could use that to shade by branch I'd be all over it.
